receiver/awscontainerinsightreceiver/internal/k8sapiserver/prometheus_scraper.go

```diff
@@ -81,10 +95,10 @@ func NewPrometheusScraper(ctx context.Context, telemetrySettings component.Telem
 	}, nil
 }
 
-func (ps *PrometheusScraper) Start() {
-	ps.simplePrometheusReceiver.Start(ps.ctx, ps.host)
+func (ps *PrometheusScraper) Start() error {
```

Comment:
Does the scraper need to adhere to an interface? We could add a compile time check. Go isn't great about it, but you could do
```go
var _ TheInterface = (*PrometheusScraper)(nil)
```
I think. Something like that. We have examples of it